activestorage/app/models/active_storage/blob.rb

```diff
@@ -140,6 +140,26 @@ def combine_signed_id_purposes(purpose) #:nodoc:
     def signed_id_verifier #:nodoc:
       @signed_id_verifier ||= ActiveStorage.verifier
     end
+
+    # Interpolates the custom storage key if needed and appends to it a unique_secure_token
+    def generate_unique_interpolated_secure_key(key:, record:, attachable:, length: MINIMUM_TOKEN_LENGTH)
```

Comment:
Is this method called often or only while uploading a new file?

Reply:
when uploading a new file!

Reply:
So, nothing to tell you ;-)
I was wondering if you could reduce Ruby object allocations.